java/acorn-comparison/src/main/java/com/datasqrl/ai/comparison/trace/TraceRecordingToolManager.java

```diff
@@ -8,32 +8,25 @@
 import com.datasqrl.ai.tool.RuntimeFunctionDefinition;
 import com.datasqrl.ai.tool.ToolManager;
 import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Preconditions;
 import java.io.IOException;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.Queue;
 import java.util.concurrent.CompletableFuture;
 import java.util.stream.Collectors;
 import lombok.NonNull;
 import lombok.Value;
 
-public class TraceRecordingToolManager extends BaseTraceToolManager {
+@Value
+public class TraceRecordingToolManager implements ToolManager {
 
-  private final Trace.TraceBuilder traceBuilder;
-  private final Queue<FunctionResponse> replayResponses;
+  @NonNull ToolManager manager;
+  @NonNull Trace.TraceBuilder traceBuilder;
+  @NonNull Optional<Trace> replayTrace;
```

Comment:
@mbroecheler 
If I understand it correctly, this TraceRecordingToolManager can work in 2 ways:
- as a recorder, when the replayTrace is empty
- as a replayer, when the replayTrace is full

This is a bit complicated, why not separate these 2 functionalities into 2 classes? I don't see a use case where we record and replay the trace in the same application.

Reply:
When we do an evaluation, we replay a gold-standard trace and record a new trace at the same time. We then compare the new trace against the gold-standard.